src/ameba/ast/visitors/implicit_return_visitor.cr

```diff
@@ -127,14 +127,13 @@ module Ameba::AST
         node.block_arg.try &.accept(self)
       end
 
-      if (return_type = node.return_type).is_a?(Crystal::Path)
+      case
+      when node.name == "initialize",
+           node.return_type.as?(Crystal::Path).try(&.names.join("::").in?("::Nil", "Nil"))
         # Special case of the return type being nil, meaning the last
         # line of the method body is ignored
-        if return_type.names.join("::").in?("::Nil", "Nil")
-          node.body.accept(self)
-        else
-          incr_stack { node.body.accept(self) }
-        end
+        # Last line of initialize methods are also ignored
+        swap_stack { node.body.accept(self) }
```

Comment:
can this change affect other rules which use this visitor?

Reply:
Yes, this change is meant to be useful for all the rules that use this visitor.

Reply:
That's fair, I can do that. I feel like this would be best covered by tests for the visitor itself though, which I can start working on soon.

Reply:
I'd suggest adding test coverage to the visitor in question in another PR.

Reply:
That was my plan. Small bite size pieces

Reply:
New release ain't gonna happen before that anyway, so IMO there's nothing to worry leaving this as-is and doing another PR solely with spec coverage.